### PR TITLE
Add gov.uk to the scraping list

### PR DESCRIPTION
**Pull request by SamDepardieu**
This PR adds the gov.uk website to the scraping list ✨ 

It also remove the legacy part of the PDF parser module, some useless comments on the who spider and add some quotes around the href attribute of the nice spider responses.

It may look a bit long, but you can safely ignore the changes in the pdf_parser module, and the structure of the gov_spider is the same as the other three spiders 👍

### CODE REVIEW

File: pdf_parser/pdf_parse.py [hunk 1]
```diff
@@ -125,10 +28,11 @@ def parse_pdf_document_pdftxt(document):
         with open(os.devnull, 'w') as FNULL:
             subprocess.check_call(cmd, stdout=FNULL)
     except subprocess.CalledProcessError as e:
+        err = e.sdterr if e.stderr else ''
         logger.warning(
             "The pdf [%s] could not be converted: %s",
             document.name,
-            e.sdterr,
```
Comment (alexwlchan) [inline pdf_parser/pdf_parse.py]: What is `e.stderr` if it's not truth-y? Is there a difference between an empty string and None?

Perhaps changing the logging token to be `%r` (repr, aka unambiguous representation of an object) would be simpler and make for better debugging?

File: wsf_scraping/spiders/gov_spider.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+
+class GovSpider(BaseSpider):
+    name = 'gov_uk'
+    custom_settings = {
+        'JOBDIR': 'crawls/gov_uk'
+    }
+
+    def start_requests(self):
+        """ This sets up the urls to scrape for each years."""
```
Comment (alexwlchan) [inline wsf_scraping/spiders/gov_spider.py]: "for each years"?

For each year's what? This isn't very clear.

File: wsf_scraping/spiders/gov_spider.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                    errback=self.on_error,
+                )
+            else:
+                self.logger.info('Not a PDF, aborting (%s)', response.url)
+                return
+
+        # Download PDF file to /tmp
+        filename = urlparse(response.url).path.split('/')[-1]
+        if filename:
+            with open('/tmp/' + filename, 'wb') as f:
```
Comment (alexwlchan) [inline wsf_scraping/spiders/gov_spider.py]: Better than using `/tmp`: use the tempfile library to ensure truly random names and no collisions.

File: wsf_scraping/spiders/gov_spider.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                    url=response.request.url,
+                    callback=self.parse,
+                    errback=self.on_error,
+                )
+            else:
+                self.logger.info('Not a PDF, aborting (%s)', response.url)
+                return
+
+        # Download PDF file to /tmp
+        filename = urlparse(response.url).path.split('/')[-1]
```
Comment (alexwlchan) [inline wsf_scraping/spiders/gov_spider.py]: Prefer using `os.path.basename` to string manipulation.

File: pdf_parser/pdf_parse.py (hunk 1 above)
Comment (SamDepardieu) [reply pdf_parser/pdf_parse.py]: Maybe `%r` would work. I though e.stderr was the best solution, but it seems that sometimes pdftohtml crashs without sending a stderr for no apparent reasons, and it leads to an error when I try to display it.
Will try with `%r` 👍

File: wsf_scraping/spiders/gov_spider.py (hunk 3 above)
Comment (SamDepardieu) [reply wsf_scraping/spiders/gov_spider.py]: In some case, I want to keep the file and the file name, should I still use the tempfile library to get the /tmp folder?

**Comment by nsorros** [comment]
as much as i prefer having fewer PR to review, i think it would be better in the future to separate such PR into two as they address different things.

again i remove liz and jonny as we agreed that they review only one PR. you can still mention them so they get notified but let's agree that assigning them means that a code review is expected

File: pdf_parser/pdf_parse.py (hunk 1 above)
Comment (nsorros) [reply pdf_parser/pdf_parse.py]: plus don't you have this information in a variable above? `err`

File: wsf_scraping/spiders/gov_spider.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                errback=self.on_error,
+            )
+
+    def save_pdf(self, response):
+        """ Retrieve the pdf file and scan it to scrape keywords and sections.
+        """
+        is_pdf = self._check_headers(response.headers)
+
+        if not is_pdf:
+            if self._check_headers(response.headers, b'text/html'):
```
Comment (nsorros) [inline wsf_scraping/spiders/gov_spider.py]: would that be cleaner if there was an `is_html` variable and you were doing 

```
if is_html statement
  parse

if not is_pdf:
   log not pdf
```

File: wsf_scraping/spiders/nice_spider.py [hunk 6]
```diff
@@ -141,11 +141,11 @@ def parse_article(self, response):
 
         if not href:
             # Second case: PDF is a single footer download link
-            href = response.css('.track-link::attr(href)').extract_first()
+            href = response.css('.track-link::attr("href")').extract_first()
```
Comment (nsorros) [inline wsf_scraping/spiders/nice_spider.py]: why does the quote change?

**Pull request merged**